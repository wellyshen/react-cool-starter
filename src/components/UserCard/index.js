/* @flow */

import React from 'react';
import type { Element } from 'react';
import styles from './styles.scss';

type Props = { info: Object };

const UserCard = ({ info }: Props): Element<'div'> => (
  <div className={styles.UserCard}>
    <h4>User Card</h4>
    <ul>
      <li>Name: {info.name}</li>
      <li>Phone: {info.phone}</li>
      <li>Email: {info.email}</li>
      <li>Website: {info.website}</li>
    </ul>
  </div>
);

export default UserCard;
