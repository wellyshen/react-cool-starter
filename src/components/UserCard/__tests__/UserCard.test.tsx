import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UserCard from '../index';

describe('<UserCard />', () => {
  it('renders', () => {
    const mockData = {
      name: 'Welly',
      phone: '007',
      email: 'test@gmail.com',
      website: 'www.test.com',
    };
    const tree = renderer
      .create(
        <MemoryRouter>
          <UserCard info={mockData} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
