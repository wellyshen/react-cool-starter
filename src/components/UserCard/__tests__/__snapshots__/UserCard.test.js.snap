// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> renders 1`] = `
<div
  className="UserCard"
>
  <h4>
    User Card
  </h4>
  <ul>
    <li>
      Name: 
      Welly
    </li>
    <li>
      Phone: 
      007
    </li>
    <li>
      Email: 
      test@gmail.com
    </li>
    <li>
      Website: 
      www.test.com
    </li>
  </ul>
</div>
`;
